ccl/workloadccl: TestSetup/gen=roachmart/flags=[--users=10_--orders=100]/batchSize=100 failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/26a06a661e7c1ce63f3077269005472182286ca4

Parameters:
```
TAGS=
GOFLAGS=
```

Stress build found a failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=595936&tab=buildLog

```
    	workload_test.go:69: pq: the batch experienced mixed success and failure
    		failed insert into orders
    		github.com/cockroachdb/cockroach/pkg/workload.Setup.func1.1
    			/go/src/github.com/cockroachdb/cockroach/pkg/workload/workload.go:306
    		github.com/cockroachdb/cockroach/pkg/workload.Setup.func1
    			/go/src/github.com/cockroachdb/cockroach/pkg/workload/workload.go:339
    		github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1
    			/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:58
    		runtime.goexit
    			/usr/local/go/src/runtime/asm_amd64.s:2361


ERROR: exit status 1

2478 runs completed, 1 failures, over 3m52s
context canceled
make: *** [stress] Error 1
Makefile:848: recipe for target 'stress' failed
```

--- a/pkg/ccl/workloadccl/workload_test.go
+++ b/pkg/ccl/workloadccl/workload_test.go
@@ -23,7 +23,6 @@ import (
 
 func TestSetup(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	t.Skip("#24655")
 
 	get := func(name string) workload.Meta {
 		meta, err := workload.Get(name)
@@ -41,7 +40,7 @@ func TestSetup(t *testing.T) {
 	}{
 		{
 			meta:        get("roachmart"),
-			flags:       []string{"--users=10", "--orders=100"},
+			flags:       []string{"--users=10", "--orders=100", "--partition=false"},
 			batchSize:   100,
 			concurrency: 4,
 		},

